### PR TITLE
feat: add function list to auto-complete to Clickhouse datasource

### LINKED ISSUE #15477: [Clickhouse] [Feature Request]  SQL editor autocomplete for functions and args
I saw that this is currently only supported in hive connections (see below for reference).

This feature is important for data exploration purposes 

https://github.com/apache/superset/blob/e466066229ec8262583bc118af1f39e2dbf77bcf/superset/db_engine_specs/hive.py#L508

### PR DESCRIPTION
**Pull request by Slach**
### SUMMARY
add function list to auto-complete to Clickhouse datasource
Fixes #15477

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TESTING INSTRUCTIONS
use https://gist.github.com/Slach/ad926dfa30ceb4fe1d168c00c3da98e0#file-bootstrap-dev-sh 
`bootstrap-dev.sh` script
open https://localhost:8088
add `clickhouse+native://default@clickhouse:9000/default` as new datasource
after add, open SQLlab and try to use auto-complete

### CODE REVIEW

File: superset/db_engine_specs/clickhouse.py [hunk 1]
```diff
@@ -69,3 +76,33 @@ def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
...
+    @cache_manager.cache.memoize()
+    def get_function_names(cls, database: "Database") -> List[str]:
+        """
+        Get a list of function names that are able to be called on the database.
+        Used for SQL Lab autocomplete.
+
+        :param database: The database to get functions for
+        :return: A list of function names usable in the database
+        """
+        df = database.get_df("SELECT name FROM system.functions")
```
Comment (zhaoyongjie) [inline superset/db_engine_specs/clickhouse.py:91]: Here we should catch some exceptions. If the user does not have read access to the system table, an error will be thrown.

Comment (Slach) [reply superset/db_engine_specs/clickhouse.py]: @zhaoyongjie fixed on https://github.com/apache/superset/pull/16234/commits/afd63bd1215848641a87c41bea7c62a5144c4fac

**Comment by zhaoyongjie** [comment]
Hi @Slach, Thanks for the following updates. LGTM excepts CI didn't pass. Could you have got time to fix it?

**Comment by Slach** [comment]
@zhaoyongjie try to fix CI/CD https://github.com/apache/superset/pull/16234/commits/33cbd2585cc500c038b3545eeaf677fa04f14186

**Comment by zhaoyongjie** [comment]
> @zhaoyongjie try to fix CI/CD [33cbd25](https://github.com/apache/superset/commit/33cbd2585cc500c038b3545eeaf677fa04f14186)
> how to trigger CD/CD workflow?

I will trigger the CI. thanks!

**Comment by Slach** [comment]
@zhaoyongjie sorry, i forgot apply black formatter
fixed on https://github.com/apache/superset/pull/16234/commits/368c2a62d9f977b55870be23df6d87ae1d806d2b

**Comment by Slach** [comment]
@zhaoyongjie ;) i also apply `isort` to clickhouse.py
https://github.com/apache/superset/pull/16234/commits/2e003cea822188fa026b40cc31b5b3be54cfa98f
could you trigger ci/cd again?

**Comment by zhaoyongjie** [comment]
> @zhaoyongjie ;) i also apply `isort` to clickhouse.py
> [2e003ce](https://github.com/apache/superset/commit/2e003cea822188fa026b40cc31b5b3be54cfa98f)
> could you trigger ci/cd again?

Sure.

**Comment by Slach** [comment]
@zhaoyongjie could you suggest about cypress-matrix tests?
https://github.com/apache/superset/pull/16234/checks?check_run_id=3398189875
look like it not related to my changes

**Comment by zhaoyongjie** [comment]
> @zhaoyongjie could you suggest about cypress-matrix tests?
> https://github.com/apache/superset/pull/16234/checks?check_run_id=3398189875
> look like it not related to my changes

Cypress might be stuck in unknown reason. I will restart this job.

**Comment by Slach** [comment]
@zhaoyongjie looks like all checks passed, could we merge pull request?

**Review by zhaoyongjie — APPROVED** [review]
image

LGTM, thanks for the contribution!

**Pull request merged**